runtime/jcl/common/reflecthelp.c

```diff
@@ -1720,3 +1720,225 @@ getFieldsHelper(JNIEnv *env, jobject cls)
 	vmFuncs->internalExitVMToJNI(vmThread);
 	return result;
 }
+
+/* Determine whether currentMethod is the accessor method for a record component with name componentName.
+ * an accessor method will have the same name as the component and take zero parameters.
+ */
+static U_8
+isRecordComponentAccessorMethodMatch(J9Method *currentMethod, const char* componentName) {
+	J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(currentMethod);
+	J9UTF8 *methodNameUtf8 = J9ROMMETHOD_NAME(romMethod);
+	U_16 methodLength = J9UTF8_LENGTH(methodNameUtf8);
+	const U_8 *methodName = J9UTF8_DATA(methodNameUtf8);
+	if ((methodLength == strlen(componentName)) 
+		&& (0 == strcmp((const char*)methodName, componentName))
+	) {
+		/* verify parameters. An accessor will not have any input parameters. */
+		J9UTF8 *methodSignatureUtf8 = J9ROMMETHOD_SIGNATURE(romMethod);
+		const U_8 *methodSignature = J9UTF8_DATA(methodSignatureUtf8);
+		if ((J9UTF8_LENGTH(methodSignatureUtf8) >= 2)
+			&& ('(' == methodSignature[0]) && (')' == methodSignature[1])
+		) {
+			return TRUE;
+		}
+	}
+	return FALSE;
```

Comment:
This doesn't look correct as it doesn't handle the return types of the signature.

For a Record like:
```java
 record Foo(int x)
```
having the following methods:
```java
x()I
x()Ljava/lang/Integer;
x()LFoo;
```
only the first is an accessor.

The current code would accept any of the 3 as the accessor method

Reply:
I left it without checking the return type because javac will generate an error that accessor methods with the wrong return type are invalid so it should not be possible to generate the other options normally.

I'll try it out with ASM as well.